contribs/gnofaucet/cooldown.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dgraph-io/badger/v4"
+	"github.com/pkg/errors"
+)
+
+// CooldownLimiter is a Limiter using an in-memory map
+type CooldownLimiter struct {
+	cooldownDB   *badger.DB
+	cooldownTime time.Duration
+}
+
+// NewCooldownLimiter initializes a Cooldown Limiter with a given duration
+func NewCooldownLimiter(cooldown time.Duration, dbPath string) *CooldownLimiter {
+	db, err := badger.Open(badger.DefaultOptions(dbPath))
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Since this error can simply occur due to bad user input, can you wrap it using `fmt.Errorf` for example to add a bit of context then return it instead of panicking?

Reply:
done here 022e92332550c136f3c7f1dbebcd01ad89529ab4